### ISSUE TITLE
portability: allow `.js` in import identifiers.

### ISSUE THREAD

**Issue by trusktr** [AssemblyScript/assemblyscript#2066]
It is awkward, but TypeScript officially supports using `.js` in import specifiers, like so:

```ts
import {foo} from './path/to/some/file.js'
console.log(foo)
```

where `./path/to/some/file.js` does not actually exist, but the file `./path/to/some/file.ts` actually exists, and TypeScript will correctly (officially) use `file.ts`. In fact, more recently, TypeScript even added the feature that when you auto-import something, if you are already using `.js` extensions in all you import specifiers, the automatically-added import statement will also have the `.js` extension for consistency.

TypeScript team has said they do not (currently at least) wish to modify import specifiers.

The reason this topic is important is that, when compiling TYpeScript code to JavaScript, we want it to work out of the box in _native_ ES Module systems (like in browsers). This requires having the `.js` extensions, because browsers do not yet support things like `import 'some/file'`, because it will try to load the equivalent of `'current-file-folder/some/file'` instead of `'current-file-folder/some/file.js'` and it will fail (unless the server is specifically configured to automatically find the matching file with the added extension, but this isn't default behavior in the average static file server).

This feature would allow us to compile code to WebAssembly, or to plain JavaScript with native ES Modules more easily (without needing an extra separate step that adds all the `.js` extensions).

**Comment by MaxGraey** [comment]
It seems TS support import `.js` files only with `--allowJs` option. And this is quite obvious, since TS is trying to smooth the migration from JavaScript to TypeScript and such features are quite appropriate in TS.

**Comment by trusktr** [comment]
TS supports `.js` extensions in TS files like this:

```ts
// This file is "foo.ts"

export const foo = 123
```

```ts
// This sibling file imports foo.ts like this:

import {foo} from './foo.js' // totally valid

console.log(foo)
```

**Comment by trusktr** [comment]
I'm having a difficult time writing portable code without this feature. Node.js requires `.js` extensions on import specifiers in modern ESM mode, but AssemblyScript can not handle the specifiers.

So, when my `index-js.ts` imports `index-wasm.ts`, and `index-wasm.ts` has something like `import {foo} from './foo'` Node.js crashes. But if I change it to `import {foo} from './foo'` then AssemblyScript crashes.

**Comment by trusktr** [comment]
> It seems TS support import `.js` files only with `--allowJs` option. And this is quite obvious, since TS is trying to smooth the migration from JavaScript to TypeScript and such features are quite appropriate in TS.

No, TS supports this for regular TypeScript, with `allowJs` and `checkJs` both turned off. It is the standard way to output ESM from TypeScript code.

**Comment by taybin** [comment]
I'm running into this with my library that I am trying to compile with AssemblyScript.
